### ISSUE TITLE
Set up clang config

### ISSUE THREAD

**Issue by php1ic** [martukas/nuclei#17]
Is there any interest to set up configuration for either, or both, of [clang-format](https://clang.llvm.org/docs/ClangFormat.html) and [clang-tidy](https://clang.llvm.org/extra/clang-tidy/)?

I'm happy to start work on this, but don't think it's a quick task, so if the project is not interested in implementing I wont.

**Comment by ferdymercury** [comment]
My two cents: I don't consider it prioritary

**Comment by php1ic** [comment]
Agreed, they are mainly quality of life changes and low priority, but definitely nice things to have.

I think the main issue is that both configs have a lot of options to decide on. Other projects appear to use inbuilt configs and make tweaks as required.

Suggested method with clang-format is to squash all changes into a single commit then add commit hash to a git config file to not steam-roll git blame output - [link](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)
